File: UsbSerialConsole/app/src/main/java/jp/sugnakys/usbserialconsole/settings/LicenseDialogFragment.java
package jp.sugnakys.usbserialconsole.settings;

import android.annotation.SuppressLint;
import android.app.AlertDialog;
import android.app.Dialog;
import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.webkit.WebView;
import android.webkit.WebViewClient;

import androidx.annotation.NonNull;
import androidx.fragment.app.DialogFragment;

import java.util.Objects;

import jp.sugnakys.usbserialconsole.R;

public class LicenseDialogFragment extends DialogFragment {

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());

        LayoutInflater inflater = (LayoutInflater) Objects.requireNonNull(getActivity()).getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        assert inflater != null;
        @SuppressLint("InflateParams") View content = inflater.inflate(R.layout.license_view_main, null);

        WebView webView = content.findViewById(R.id.webview);
        webView.setWebViewClient(new WebViewClient());
        webView.loadUrl("file:///android_asset/license/license.html");

        builder.setTitle(getString(R.string.license_title));
        builder.setView(content);

        return builder.create();
    }
}

File: UsbSerialConsole/app/src/main/java/jp/sugnakys/usbserialconsole/settings/LicenseDialogFragment.kt
package jp.sugnakys.usbserialconsole.settings

import android.app.AlertDialog
import android.app.Dialog
import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.webkit.WebView
import android.webkit.WebViewClient
import androidx.fragment.app.DialogFragment
import jp.sugnakys.usbserialconsole.R

class LicenseDialogFragment : DialogFragment() {

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val inflater = activity?.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        val content = inflater.inflate(R.layout.license_view_main, null)

        val webView = content.findViewById<WebView>(R.id.webview)
        webView.webViewClient = WebViewClient()
        webView.loadUrl("file:///android_asset/license/license.html")

        return AlertDialog.Builder(activity)
                .setTitle(getString(R.string.license_title))
                .setView(content)
                .create()
    }
}
